src/HelpersServiceProvider.php

```diff
@@ -76,12 +75,20 @@ protected function extendValidator(): void
             $keyField = Arr::get($parameters, 1, 'id');
 
             if (!empty(Arr::get($parameters, 2))) {
-                $value = collect($value)->pluck(Arr::get($parameters, 2));
+                $value = collect($value)
+                    ->pluck(Arr::get($parameters, 2))
+                    ->toArray();
             }
 
-            return DB::table($table)
+            $value = array_unique($value);
+
+            $existingValue = DB::table($table)
                 ->whereIn($keyField, $value)
-                ->exists();
+                ->distinct()
+                ->pluck($keyField)
+                ->toArray();
```

Comment:
```suggestion
                ->distinct()
                ->count();
```

Reply:
We have to use `pluck($keyField)` before `distinct()` because without it all the entries will be unique

Reply:
You are right. 

We can modify the code to:
```
DB::table($table)
                ->whereIn($keyField, $value)
                ->distinct()
                ->count($keyField);
```

In this case `distinct` will work correctly with `count`